pkg/controller/component/vars.go

```diff
@@ -381,15 +381,15 @@ func resolveClusterObjectRefVars(ctx context.Context, cli client.Reader, synthes
 	vars1, vars2 := make([]corev1.EnvVar, 0), make([]corev1.EnvVar, 0)
 	for _, v := range definedVars {
 		switch {
-		case len(v.Value) > 0:
-			vars1 = append(vars1, corev1.EnvVar{Name: v.Name, Value: v.Value})
 		case v.ValueFrom != nil:
-			var1, var2, err := resolveClusterObjectVarRef(ctx, cli, synthesizedComp, v.Name, *v.ValueFrom)
+			var1, var2, err := resolveClusterObjectVarRef(ctx, cli, synthesizedComp, v.Name, *v.ValueFrom, v)
```

Comment:
Seems "resolveClusterObjectRefVars" is better renamed to "resolveClusterObjectVars"

Reply:
Otherwise resolveClusterObjectRefVars and resolveClusterObjectVarRef has the same "resolveClusterObject" prefix, but their suffix "RefVars" and "VarRef" is hard to distinguish.